Makefile

```diff
@@ -1,12 +1,21 @@
+# Gets all packages that include a Rust crates
 WORKSPACES_WITH_RUST_MODULES := $(wildcard $(addsuffix /crates, $(wildcard ./packages/*)))
 
+# Gets all individual crates such that they can get built
+CRATES := $(foreach crate,${WORKSPACES_WITH_RUST_MODULES},$(dir $(wildcard $(crate)/*/Cargo.toml)))
+
 .POSIX:
 
 all: help
 
-.PHONY: $(WORKSPACES_WITH_RUST_MODULES) ## build all WASM modules
+.PHONY: $(CRATES) ## builds all Rust crates
+$(CRATES):
+# --out-dir is relative to working directory
+	wasm-pack build --target=bundler --out-dir ./pkg $@
```

Comment:
This will execute `wasm-pack build` for each of the crate individually. I though the idea was to make this all one in go and let wasm-pack decide how to optimize?

Reply:
1`Exactly, this initiates multiple top-level calls of `wasm-pack`, namely e.g. `wasm-pack build ./packages/crates/some-crate`.

Since `wasm-pack` as of now, does not support cargo workspaces, each crate need to be build individually. The main enhancement is that `wasm-pack` is called from top-level, such that the underlying `cargo build` call evaluates its build-cache in `./target` whereas, when called in `./packages/some-package/crates/some-crate` the underlying `cargo build` call appear to be unaware of potential other builds and will rebuild *all* crates, although some of them have already been built.